src/combinatorics/set_family/compression/uv.lean

```diff
@@ -355,11 +355,15 @@ begin
     { refine or.inl ⟨hs, _⟩,
       rw compress,
       split_ifs with huvs,
-      { by_contra hsuv,
-        refine (H _ _ hsuv).2.2.2 _,
-        { sorry },
-        rwa [sdiff_union_self_eq_union, union_eq_left_iff_subset.2 (le_sup_of_le_left huvs.2),
-          huvs.1.symm.sup_sdiff_cancel_right] },
+      { rcases huvs with ⟨hus, hvs⟩,
+        rw [mem_shadow_iff] at hs',
+        rcases hs' with ⟨t, Ht, a, Hat, ta⟩,
+        have hav : a ∉ v := sorry, -- by hvs and the fact that a ∉ s
+        have ht : t ∈ 𝒜 := sorry, -- by mem_of_mem_compression and Ht
+        have ht' : (t ⊔ u) \ v ∈ 𝒜 := sorry, -- by sup_sdiff_mem_of_mem_compression and Ht
```

Comment:
i think (at least the way you suggested proving this) this needs `a ∉ u`

Reply:
which I'm not sure follows. i have sorry free proofs of hav and ht

Reply:
similarly I think the same for the last sorry

Reply:
You're right, a different argument was needed for the case `a \in U`. Fixed.

Reply:
I'm happy to hack away at the sorries too, do you want me to do that or do you want to do it yourself? :)

Reply:
I think the proof will need *a lot* of golfing from more experienced people (such as yourself). Unless I missed something, the remaining sorries should be really math-trivial, which means any proofs I write will also need to be heavily golfed, so help is greatly appreciated :) We should just check if @YaelDillies isn't already working on them to avoid duplication.

Reply:
Feel free to! I'm whipping other cats right now.